disco_aws_automation/disco_alarm_config.py

```diff
@@ -206,15 +226,23 @@ def _decode_section_name(section):
         Extract special alarm parameters which are stored in section name
         """
         section_segments = section.split(".")
-        if len(section_segments) == 4:
-            team, namespace, metric_name, section_hostclass = section_segments
+        # If a section has four or more segments, treat the middle segments as the metric name and
+        # the ending segment as the hostclass. This is to support some Elasticsearch metrics like
+        # 'ClusterStatus.red'
```

Comment:
How about using a different delimiter for metric_name part of the section. e.g. `ClusterStatus_red` or `ClusterStatus-red`?